### DISCUSSION TITLE
How to Reproduce the Full Pretraining

### DISCUSSION THREAD

**Discussion by vzmath** [tbepler/topaz#150] [General]
Hi, 

Thanks very much for providing the very-well documented code. I have a general question after reading the Topaz-Denoise: general deep denoising models for cryoEM and cryoET and going over the tutorials. 

In particular, if one would like to reproduce the full training for U-net on Large dataset (both discussed in your paper supplementary) from scratch, what are the steps? Is there a short code snippet that could demonstrate the process? 

Your feedback would be greatly appreciated!

**Comment by tbepler** [comment]
You can reproduce the training using the `topaz denoise` command. To train, set the `-a/--dir-a` argument to a directory containing the odd half micrographs and the `-b/--dir-b` argument to a directory containing the even half micrographs.

We don't have a way to download the processed micrographs, but you can download all of the NYSBC data used to train [here](https://www.ebi.ac.uk/empiar/EMPIAR-10473/) and the other EMPIAR datasets listed in the manuscript to reproduce the full training. For the datasets with >200 micrographs, we randomly sampled 200 (unfortunately, I'm not sure I have a record of the specific micrographs we sampled in these cases... I'll double check, but it shouldn't make a difference for reproducing the training). These were all preprocessed by splitting the movie frames into two micrographs, one for the odd frames and one for the even frames, and then aligning and summing with motioncor2 using 5 by 5 patches and a b-factor of 100.

Hope that helps!
